xbmc/utils/SystemInfo.cpp

```diff
@@ -508,9 +508,17 @@ std::string CSysInfo::GetKernelVersionFull(void)
     return kernelVersionFull;
 
 #if defined(TARGET_WINDOWS_DESKTOP)
-  OSVERSIONINFOEXW osvi;
+  OSVERSIONINFOEXW osvi = {};
+  DWORD buildRevision = 0;
+  DWORD len = sizeof(DWORD);
+
+  // get UBR (updates build revision)
+  RegGetValueW(HKEY_LOCAL_MACHINE, L"Software\\Microsoft\\Windows NT\\CurrentVersion", L"UBR",
```

Comment:
Does this key exist on all supported Windows versions (>=7)?

Reply:
Yep!

![Anotación 2020-08-16 102310](https://user-images.githubusercontent.com/58434170/90330132-b6a62e80-dfaa-11ea-8317-6fc88b0bac19.png)

Reply:
OK, so we basically assume that the call cannot fail, right?

Reply:
Well, UBR key is supported in Win 7, 8, 8.1 and 10. Anyway if fails (for unexpected reason) nothing bad happens:

`DWORD buildRevision = 0;`

Stays at 0 and kernelVersion string it would look like e.g. `10.0.19041.0`

Reply:
I would prefer if it would just omit the last part in that case (if future windows versions don't have UBR)

Reply:
It's an unlikely scenario, but I agree that it's better this way. I've already changed it.